lib/FreeRTOS-Plus-TCP/source/FreeRTOS_TCP_IP.c

```diff
@@ -1158,162 +1171,183 @@ UBaseType_t uxNewMSS;
 
 	/* The comparison with pucLast is only necessary in case the option data are
 	corrupted, we don't like to run into invalid memory and crash. */
-	while( pucPtr < pucLast )
+	should_continue_loop = pdTRUE;
+	while( ( pucPtr < pucLast ) && ( should_continue_loop == pdTRUE ) )
 	{
-		UBaseType_t xRemainingOptionsBytes = pucLast - pucPtr;
+		should_continue_loop = prvSingleStepTCPHeaderOptions( &pucPtr, &pucLast, &pxSocket, &pxTCPWindow );
+	}
+}
 
-		if( pucPtr[ 0 ] == TCP_OPT_END )
-		{
-			/* End of options. */
-			break;
-		}
-		if( pucPtr[ 0 ] == TCP_OPT_NOOP)
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvSingleStepTCPHeaderOptions( const unsigned char ** const pucPtr, const unsigned char ** const pucLast, FreeRTOS_Socket_t ** const pxSocket, TCPWindow_t ** const pxTCPWindow)
+{
+	UBaseType_t uxNewMSS;
+	UBaseType_t xRemainingOptionsBytes = ( *pucLast ) - ( *pucPtr );
+
+	if( ( *pucPtr )[ 0 ] == TCP_OPT_END )
+	{
+		/* End of options. */
+		return pdFALSE;
```

Comment:
The FreeRTOS docs say that we are using a single-point-of-return and that there are only 2 deviations. (https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html) 

I think we should consider refactoring to maintain the single point of return, or discuss if this requires an update to documentation if we opt for replacing breaks with return pdFALSE.